server/events/apply_requirement_handler.go

```diff
@@ -0,0 +1,50 @@
+package events
+
+import (
+	"github.com/pkg/errors"
+	"github.com/runatlantis/atlantis/server/core/runtime"
+	"github.com/runatlantis/atlantis/server/events/models"
+	"github.com/runatlantis/atlantis/server/events/yaml/raw"
+	"github.com/runatlantis/atlantis/server/events/yaml/valid"
+)
+
+//go:generate pegomock generate -m --package mocks -o mocks/mock_apply_handler.go ApplyRequirementHandler
+type ApplyRequirementHandler interface {
```

Comment:
Sort of an ambiguous name imo.  What this is really doing is validating the current state against the required state right?

Additionally, I'd imagine that we could eventually take this further and have this interface represent a single apply requirement implementation altogether (instead of all of them).

Therefore, something like `ApplyRequirement` as an interface makes more sense to me with a method named something like`validateProject(...) error {...}`.

Reply:
Done!